iceoryx_posh/test/moduletests/test_roudi_process_manager.cpp

```diff
@@ -114,3 +115,29 @@ TEST_F(ProcessManager_test, RegisterAndUnregisterWorks)
 
     EXPECT_TRUE(unregisterResult);
 }
+
+TEST_F(ProcessManager_test, HandleProcessShutdownPreparationRequestWorks)
+{
+    m_sut->registerProcess(m_processname, m_pid, m_user, m_isMonitored, 1U, 1U, m_versionInfo);
+
+    auto user = iox::posix::PosixUser::getUserOfCurrentProcess().getName();
+    auto payloadDataSegmentMemoryManager =
+        m_roudiMemoryManager->segmentManager().value()->getSegmentInformationForUser(user).m_memoryManager;
+
+    // get publisher and subscriber
+    PublisherOptions publisherOptions{
+        0U, iox::NodeName_t("node"), true, iox::popo::SubscriberTooSlowPolicy::WAIT_FOR_SUBSCRIBER};
+    PublisherPortUser publisher(
+        m_portManager
+            ->acquirePublisherPortData(
+                {1U, 1U, 1U}, publisherOptions, m_processname, payloadDataSegmentMemoryManager, PortConfigInfo())
+            .value());
+
+    ASSERT_TRUE(publisher.isOffered());
+
+    m_sut->handleProcessShutdownPreparationRequest(m_processname);
+
```

Comment:
well, in this case the app might be again blocked. It's the responsibility of the application developer to not do this since we have no/less chance to prevent this